File: test/codegen/tree_shaking2.dart
abstract class Foo {
  foo() => 1;
  operator[](x);
  bar() {}
}

class Bar extends Foo {
  foo() => 2;
  operator[](x) => 2;
  bar() {}
}

class Baz extends Foo {
  foo() => 3;
  operator[](x) => 3;
}

class Constants {
  static final X = 1;
  static final Y = 2;
}

main() {
  Foo foo;// = new Foo();
  foo = new Bar();
  print(foo.foo());
  print(foo[1]());
  print(Constants.X);
}

File: test/codegen/expect/tree_shaking2.js
dart_library.library('tree_shaking2', null, /* Imports */[
  'dart/_runtime',
  'dart/core'
], /* Lazy imports */[
], function(exports, dart, core) {
  'use strict';
  let dartx = dart.dartx;
  class Foo extends core.Object {
    foo() {
      return 1;
    }
  }
  dart.setSignature(Foo, {
    methods: () => ({foo: [dart.dynamic, []]})
  });
  class Bar extends Foo {
    foo() {
      return 2;
    }
    get(x) {
      return 2;
    }
  }
  dart.setSignature(Bar, {
    methods: () => ({get: [dart.dynamic, [dart.dynamic]]})
  });
  class Constants extends core.Object {}
  Constants.X = 1;
  function main() {
    let foo = null;
    foo = new Bar();
    core.print(foo.foo());
    core.print(dart.dcall(foo.get(1)));
    core.print(Constants.X);
  }
  dart.fn(main);
  // Exports:
  exports.Foo = Foo;
  exports.Bar = Bar;
  exports.Constants = Constants;
  exports.main = main;
});
